notty_lwt.opam

```diff
@@ -0,0 +1,18 @@
+opam-version: "1.2"
+homepage:     "https://github.com/pqwy/notty"
+dev-repo:     "https://github.com/pqwy/notty.git"
+bug-reports:  "https://github.com/pqwy/notty/issues"
+author:       "David Kaloper <david@numm.org>"
+maintainer:   "David Kaloper <david@numm.org>"
+license:      "ISC"
+
+build: [
+  ["jbuilder" "subst"] {pinned}
+  ["jbuilder" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "jbuilder" {build}
```

Comment:
could we remove the `{build}` here, so `notty` won't end up in an unfixable state next time `jbuilder` breaks something like it happened with `0.18`?

Reply:
Not sure I understand; could you elaborate your comment?

AFAICS, `{build}` is the right thing here - a new `jbuilder` version should not cause `notty` to be re-installed.

Reply:
@nojb IMHO, after spending a few frustrating hours trying to get `utop` to work last time `jbuilder` broke that, and the `{build}` tag caused `utop` to **stay** broken even after a new revision of `jbuilder` was installed on the system, I disagree: I think a new `jbuilder` version should cause a new compile.

Reply:
Am not familiar with the particular `utop` issue, but sounds to me like it was a one-off bug that was easily fixed by forcing its reinstallation after updating `jbuilder`.

On the other hand, if we followed your suggestion of declaring a (non-build) dependencies on `jbuilder`, any new release of `jbuilder` would cause large parts of your installed `opam` universe to be rebuilt -- clearly not what one would like in general.

Reply:
@nojb I'm not sure how I was supposed to easily force reinstallation, for me it was pretty hard. It's not like the fixed version of `jbuilder` had a message about it either, so it took a while to find the github issue threads where people were complaining about it. At any rate, for me a "one off bug" in a released version of a project that is still in `beta` sounds to me like it could easily happen again, and thus I think it makes sense to take that hit while `jbuilder` is still unstable.

I agree that is is not ideal, and rebuilding `world` should only be done when a new revision fixes a bug, but unfortunately it doesn't seem like anyone has come up with a trick to do that with `opam`.